### PR TITLE
CORS-3580: Add the hyperdisk-balanced disk type

### PR DESCRIPTION
**Pull request by barbacbd**
** Add the hyperdisk-balanced disk type for gcp machines. 
** Add the test to validate the new type.

### CODE REVIEW

**Comment by JoelSpeed** [comment]
@barbacbd Have you looked at the GCP provider code itself to see if it's doing any other validation?

**Comment by barbacbd** [comment]
> @barbacbd Have you looked at the GCP provider code itself to see if it's doing any other validation?

@JoelSpeed I did. It looked like the only mention of the other disk types was in the vendor code.

File: pkg/webhooks/machine_webhook_test.go [hunk 1]
```diff
@@ -3365,7 +3365,7 @@ func TestValidateGCPProviderSpec(t *testing.T) {
 				}
 			},
 			expectedOk:    false,
-			expectedError: "providerSpec.disks[0].type: Unsupported value: \"invalid\": supported values: \"pd-balanced\", \"pd-ssd\", \"pd-standard\"",
+			expectedError: "providerSpec.disks[0].type: Unsupported value: \"invalid\": supported values: \"pd-balanced\", \"pd-ssd\", \"pd-standard\", \"hyperdisk-balanced\"",
```
Comment (r4f4) [inline pkg/webhooks/machine_webhook_test.go:3368]: `sets.List` returns an ordered list, so you need to make sure the support values here is ordered:
```suggestion
			expectedError: "providerSpec.disks[0].type: Unsupported value: \"invalid\": supported values: \"hyperdisk-balanced\", \"pd-balanced\", \"pd-ssd\", \"pd-standard\"",
```

**Pull request merged**